diff -ruiN chipmunk/cpBody.c chipmunk_new//cpBody.c
--- chipmunk/cpBody.c	2010-03-19 15:03:29.000000000 +0200
+++ chipmunk_new//cpBody.c	2010-03-19 15:04:11.000000000 +0200
@@ -49,6 +49,7 @@
 	cpBodySetAngle(body, 0.0f);
 	body->w = 0.0f;
 	body->t = 0.0f;
+        body->gravity = 0.0f;
 	
 	body->v_bias = cpvzero;
 	body->w_bias = 0.0f;
diff -ruiN chipmunk/cpBody.h chipmunk_new//cpBody.h
--- chipmunk/cpBody.h	2010-03-19 12:03:39.000000000 +0200
+++ chipmunk_new//cpBody.h	2010-03-19 15:04:11.000000000 +0200
@@ -53,11 +53,13 @@
 	// Angular components of motion (angle, angular velocity, and torque)
 	// Always use cpBodySetAngle() to set the angle of the body as a and rot must agree.
 	cpFloat a, w, t;
+
+    cpFloat gravity;
 	
 	// Cached unit length vector representing the angle of the body.
 	// Used for fast vector rotation using cpvrotate().
 	cpVect rot;
-	
+
 	// *** User Definable Fields
 	
 	// User defined data pointer.
diff -ruiN chipmunk/cpSpace.c chipmunk_new//cpSpace.c
--- chipmunk/cpSpace.c	2010-03-19 15:03:29.000000000 +0200
+++ chipmunk_new//cpSpace.c	2010-03-19 15:04:11.000000000 +0200
@@ -849,7 +849,8 @@
 	cpFloat damping = cpfpow(1.0f/space->damping, -dt);
 	for(int i=0; i<bodies->num; i++){
 		cpBody *body = (cpBody *)bodies->arr[i];
-		body->velocity_func(body, space->gravity, damping, dt);
+                cpVect grav = cpv(0.0, space->gravity.y + body->gravity);
+		body->velocity_func(body, grav, damping, dt);
 	}
 
 	for(int i=0; i<arbiters->num; i++)
