--- sglib-1.0.4/sglib.h	2006-08-02 11:11:02.000000000 +0300
+++ sglib.h	2009-10-13 14:50:24.000000000 +0300
@@ -383,7 +383,6 @@
 
 #define SGLIB_LIST_MAP_ON_ELEMENTS(type, list, iteratedVariable, next, command) {\
   type *_ne_;\
-  type *iteratedVariable;\
   (iteratedVariable) = (list); \
   while ((iteratedVariable)!=NULL) {\
     _ne_ = (iteratedVariable)->next;\
@@ -1276,7 +1276,7 @@
  type *sglib_##type##_it_next(struct sglib_##type##_iterator *it) {\
    type *ce, *eq;\
    int  (*scp)(type *, type *);\
-   int  c;\
+   int  c = 0;\
    ce = it->nextelem;\
    it->nextelem = NULL;\
    if (it->subcomparator != NULL) {\
